src/main/java/org/opentripplanner/transit/model/timetable/RealTimeTripTimes.java

```diff
@@ -0,0 +1,505 @@
+  @Override
+  public int getDepartureDelay(final int stop) {
+    return getDepartureTime(stop) - scheduledTripTimes.getScheduledDepartureTime(stop);
+  }
+
+  public void setRecorded(int stop) {
+    setStopRealTimeStates(stop, StopRealTimeState.RECORDED);
+  }
+
+  public void setCancelled(int stop) {
+    setStopRealTimeStates(stop, StopRealTimeState.CANCELLED);
+  }
+
+  public void setNoData(int stop) {
+    setStopRealTimeStates(stop, StopRealTimeState.NO_DATA);
+  }
+
+  public void setPredictionInaccurate(int stop) {
+    setStopRealTimeStates(stop, StopRealTimeState.INACCURATE_PREDICTIONS);
+  }
+
+  public boolean isCancelledStop(int stop) {
+    return isStopRealTimeStates(stop, StopRealTimeState.CANCELLED);
+  }
+
+  public boolean isRecordedStop(int stop) {
+    return isStopRealTimeStates(stop, StopRealTimeState.RECORDED);
+  }
+
+  public boolean isNoDataStop(int stop) {
+    return isStopRealTimeStates(stop, StopRealTimeState.NO_DATA);
+  }
+
+  public boolean isPredictionInaccurate(int stop) {
+    return isStopRealTimeStates(stop, StopRealTimeState.INACCURATE_PREDICTIONS);
+  }
+
+  public void setOccupancyStatus(int stop, OccupancyStatus occupancyStatus) {
+    prepareForRealTimeUpdates();
+    this.occupancyStatus[stop] = occupancyStatus;
+  }
+
+  /**
+   * This is only for API-purposes (does not affect routing).
+   */
+  @Override
+  public OccupancyStatus getOccupancyStatus(int stop) {
+    if (this.occupancyStatus == null) {
+      return OccupancyStatus.NO_DATA_AVAILABLE;
+    }
+    return this.occupancyStatus[stop];
+  }
+
+  @Override
+  public BookingInfo getDropOffBookingInfo(int stop) {
+    return scheduledTripTimes.getDropOffBookingInfo(stop);
+  }
+
+  @Override
+  public BookingInfo getPickupBookingInfo(int stop) {
+    return scheduledTripTimes.getPickupBookingInfo(stop);
+  }
+
+  @Override
+  public boolean isScheduled() {
+    return realTimeState == RealTimeState.SCHEDULED;
+  }
+
+  @Override
+  public boolean isCanceledOrDeleted() {
+    return isCanceled() || isDeleted();
+  }
+
+  @Override
+  public boolean isCanceled() {
+    return realTimeState == RealTimeState.CANCELED;
+  }
+
+  @Override
+  public boolean isDeleted() {
+    return realTimeState == RealTimeState.DELETED;
+  }
+
+  @Override
+  public RealTimeState getRealTimeState() {
+    return realTimeState;
+  }
+
+  public void setRealTimeState(final RealTimeState realTimeState) {
+    this.realTimeState = realTimeState;
+  }
+
+  /**
+   * When creating a scheduled TripTimes or wrapping it in updates, we could potentially imply
+   * negative running or dwell times. We really don't want those being used in routing. This method
+   * checks that all times are increasing.
+   *
+   * @throws org.opentripplanner.transit.model.framework.DataValidationException of the first error found.
+   */
+  public void validateNonIncreasingTimes() {
+    final int nStops = arrivalTimes.length;
+    int prevDep = -9_999_999;
```

Comment:
Can't we have this value as a constant?

Reply:
Yes, I will fix it when doing the final cleanup of trip-times, but first I need to cleanup the usage - leaving the internals as is. Se my next comment and the PR summary.